qcfractal/testing.py

```diff
@@ -137,6 +137,18 @@ def preserve_cwd():
         os.chdir(cwd)
 
 
+@contextmanager
+def queue_manager(client, adapter, **kwargs):
```

Comment:
I'm still wrapping my head around how `yield from` works with pytest and this is what I came up with. In order to reduce the code duplication, i made the `build_managed_compute_server` function use the `build_compute_server`, but then getting the `yield client, server, adapter_client` generator to split out into a `client, server, adapter_client = build_compute_server(mtype)` to then pass that back into another `yield client, server, manager` wasn't working well. So I made the contextmanager as a way around that

Reply:
What I mean is add a `__enter__` and `__exit__` to the QueueManager class so that you can do:
```
with QueueManager(stuff) as manager:
 ...
```

For the adapter, you want to pull out *just* the adapter bits and straight return them as their own manager context cleanup correctly. We can also consider adding a `__del__` function to QueueManager that shuts down its adapter.

Reply:
> What I mean is add a `__enter__` and `__exit__` to the QueueManager class so that you can do

I specifically meant I was trying to avoid doing that if possible, I reworked it so I don't need that anymore and am currently restoring what I broke.

Reply:
I think it would be a good thing to have in general was my main point.